pkg/minishift/network/settings.go

```diff
@@ -0,0 +1,73 @@
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package network
+
+import (
+	"bytes"
+	"fmt"
+	"text/template"
+
+	"github.com/minishift/minishift/pkg/util/os/atexit"
+)
+
+const (
+	configureIPAddressMessage      = "-- Attempting to set network settings ..."
+	configureIPAddressFailure      = "FAIL\n   not supported on this platform"
+	configureNetworkScriptTemplate = `DEVICE={{.Device}}
+IPADDR={{.IPAddress}}
+NETMASK={{.Netmask}}
+GATEWAY={{.Gateway}}
+DNS1={{.DNS1}}
+DNS2={{.DNS2}}
+`
+)
+
+type NetworkSettings struct {
+	Device    string
+	IPAddress string
+	Netmask   string
+	Gateway   string
+	DNS1      string
+	DNS2      string
+}
+
+func PrintNetworkSetting(networkSettings NetworkSettings) {
+	fmt.Println(configureIPAddressMessage)
+	fmt.Println("   Device:     ", networkSettings.Device)
+	fmt.Println("   IP Address: ", fmt.Sprintf("%s/%s", networkSettings.IPAddress, networkSettings.Netmask))
+	if networkSettings.Gateway != "" {
+		fmt.Println("   Gateway:    ", networkSettings.Gateway)
+	}
+	if networkSettings.DNS1 != "" {
+		fmt.Println("   Nameservers:", fmt.Sprintf("%s %s", networkSettings.DNS1, networkSettings.DNS2))
+	}
+}
+
+func FillNetworkSettingsScript(networkSettings NetworkSettings) string {
```

Comment:
The naming is sounding confusing to me. It seems like this function fill the network settings into script(file).

Probably, `FillNetworkSettingsInfo` or similar?

Reply:
It does become a script, but the script is unrolled in the iso.

Reply:
The idea is to re-use this in a follow-up task.
The naming was done as it is using the old networj-script structure and info fields

Reply:
@gbraad Both FillNetworkSettingsScript and PrintNetworkSetting are currently used in the same package.So I do not see any value in making them public functions. Is there any specific reason you wanted them as public functions?